Themes

shmolf opened the pull request:
Added theme support, starting with the Default and a Dark theme.

Adding new themes should be simple.
- add new references in the `styles.css` file
- add new theme in the list of themes in `preferences.html`
- add new theme in the constant at the top of `Themes.js`

Rendering may vary on MacOS (which I don't have access to), but it's Chromium, so I hope its fairly consistent.

Any file I open auto-eliminates trailing white-space, so I apologize again for the excessive white-space removals.

[Related Issue](https://github.com/thamara/time-to-leave/issues/15)

tupaschoal commented:
Nice @shmolf! Can you add some screenshots of the themes?

tupaschoal commented on js/UserPreferences.js:
```diff
@@ -14,6 +15,7 @@ const defaultPreferences = {
     'working-days-friday': true,
     'working-days-saturday': false,
     'working-days-sunday': false,
+    'theme': ''
```
You should initialize it to `Default` (or White), I guess

shmolf replied:
Yes, I missed that.

shmolf replied:
Going to set it to `light` per [this comment](https://github.com/thamara/time-to-leave/pull/69#discussion_r336736189)

tupaschoal commented on js/Themes.js:
```diff
@@ -0,0 +1,27 @@
+const themeOptions = [ 'default', 'dark' ];
```
call it `light` and `dark`

tupaschoal commented on src/preferences.html:
```diff
@@ -58,6 +58,16 @@
+        <div class="theme-setting">
+            <div class="theme-option">
+                <select id="theme" name="notification">
+                    <option value="default" selected>Default</option>
```
This selected somehow isn't working, when I first opened it, it had nothing selected.

tupaschoal commented:
One other comment, the preferences dialog doesn't follow the theme

shmolf commented:
I left the preferences as the original color in case the alternative theme doesn't render well on the users screen. In this way, the theme could always be reverted, however, I could deff set the Preferences page to the same theme.

tupaschoal commented:
What do you think @thamara?

thamara commented:
Here's what it looks like on mac:

![Screen Shot 2019-10-19 at 23 47 41](https://user-images.githubusercontent.com/846063/67154034-ee7d1080-f2ca-11e9-81e4-f39c79ee31e4.png)

I think the only thing that bothers me is the color of the punch in button...

shmolf commented:
I've renamed the Default theme to `light`.
I've fixed the Punch button for the dark them (and also the light theme?)/

Added the missing semicolon in the `main.js` file.

Themes are applied in the Preferences pane, and they are previewed immediately. I think this should solve the issue of a user who selects a theme that makes it hard to see, this way, they have an option to set it back to light before they close the pane.

If you notice in the `preferences.css` there's some additional CSS variables. These are variables specific to the Preferences page, but they could move to the main Theme file, if you decide to make those values more global.

shmolf commented:
Screen shots - Windows 7

Main  Interface - Light
![image](https://user-images.githubusercontent.com/6433103/67163385-27090280-f33c-11e9-9791-44dfd862106b.png)

Preferences Pane - Light
![image](https://user-images.githubusercontent.com/6433103/67163372-10fb4200-f33c-11e9-9904-ac443ec2b452.png)


Main Interface - Dark
![image](https://user-images.githubusercontent.com/6433103/67163354-e610ee00-f33b-11e9-88c2-cfe1df17dadd.png)

Preferences Pane - Dark
![image](https://user-images.githubusercontent.com/6433103/67163363-fcb74500-f33b-11e9-8d79-173e529337a6.png)

shmolf commented:
:_tuzki_dancing_left:
